From 7b238e793fc541ed69094d676cde817a9691dede Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Mon, 7 Mar 2022 20:58:46 +0100
Subject: [PATCH 05/16] user32/tests: Remove SetForegroundWindow success
 checks.

Although it could fail and cause spurious failures, it's not supposed to
and skipping the tests simply creates false successful runs.
---
 dlls/user32/tests/win.c | 21 +++------------------
 1 file changed, 3 insertions(+), 18 deletions(-)

diff --git a/dlls/user32/tests/win.c b/dlls/user32/tests/win.c
index 11111111111..11111111111 100644
--- a/dlls/user32/tests/win.c
+++ b/dlls/user32/tests/win.c
@@ -3500,12 +3500,7 @@ static void test_popup_zorder(HWND hwnd_D, HWND hwnd_E, DWORD style)
     HWND hwnd_A, hwnd_B, hwnd_C, hwnd_F;
 
     /* Give current thread foreground state otherwise the tests may fail. */
-    if (!SetForegroundWindow(hwnd_D))
-    {
-        skip("SetForegroundWindow not working\n");
-        return;
-    }
-
+    SetForegroundWindow(hwnd_D);
     SetWindowPos(hwnd_E, hwnd_D, 0,0,0,0, SWP_NOSIZE|SWP_NOMOVE|SWP_NOACTIVATE);
 
     check_z_order(hwnd_D, hwnd_E, 0, 0, FALSE);
@@ -3962,12 +3957,7 @@ static void test_SetForegroundWindow(HWND hwnd)
     ShowWindow(hwnd, SW_SHOW);
     check_wnd_state(hwnd, hwnd, hwnd, 0);
 
-    ret = SetForegroundWindow(hwnd);
-    if (!ret)
-    {
-        skip( "SetForegroundWindow not working\n" );
-        return;
-    }
+    SetForegroundWindow(hwnd);
     check_wnd_state(hwnd, hwnd, hwnd, 0);
 
     SetLastError(0xdeadbeef);
@@ -11747,12 +11737,7 @@ static void test_topmost(void)
     ok(owner != 0, "Failed to create owner window (%ld)\n", GetLastError());
 
     /* Give current thread foreground state otherwise the tests may fail. */
-    if (!SetForegroundWindow(owner))
-    {
-        DestroyWindow(owner);
-        skip("SetForegroundWindow not working\n");
-        return;
-    }
+    SetForegroundWindow(owner);
 
     hwnd = create_tool_window(WS_CAPTION | WS_SYSMENU | WS_MINIMIZEBOX |
                               WS_MAXIMIZEBOX | WS_POPUP | WS_VISIBLE, owner);
-- 
2.40.0

